src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -119,14 +133,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Protected Overrides Function ExpressionKind() As OperationKind
             Dim rightBinary As BoundBinaryOperator = TryCast(Me.Right, BoundBinaryOperator)
             If rightBinary IsNot Nothing Then
-                If TypeOf rightBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If rightBinary.Left.Kind = BoundKind.CompoundAssignmentTargetPlaceholder Then
                     Return OperationKind.CompoundAssignmentExpression
                 End If
             End If
 
             Dim rightOperatorBinary As BoundUserDefinedBinaryOperator = TryCast(Me.Right, BoundUserDefinedBinaryOperator)
             If rightOperatorBinary IsNot Nothing Then
-                If TypeOf rightOperatorBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If DirectCast(rightOperatorBinary, IBinaryOperatorExpression).LeftOperand.Kind = BoundKind.CompoundAssignmentTargetPlaceholder Then
```

Comment:
I don't think this compares the right Kind now. Does IOperation.Kind returns BoundKind? Either both sides should be in BoundNode universe, or in IOperation universe.